ruby/examples/lib/apply_order_discount.rb

```diff
@@ -0,0 +1,44 @@
+class ApplyOrderDiscount
+  Request = Struct.new(:id)
+  Response = Struct.new(:discount, :has_discount?)
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
```

Comment:
Could happen if you were updating the domain language... e.g. StockLevel to AvailableToPromise (https://en.wikipedia.org/wiki/Available-to-promise) as your knowledge improved, and to keep consistency between your code and the real world concepts it is representing.